scripts/priorities.py

```diff
@@ -0,0 +1,170 @@
+    d = d + (1*(sparse_matrix_A==99) * (sparse_matrix_B.transpose()==99))
+    d = d + (1*(sparse_matrix_A==103) * (sparse_matrix_B.transpose()==103))
+    d = d + (1*(sparse_matrix_A==116) * (sparse_matrix_B.transpose()==116))
+
+    d = d.todense()
+
+    n_comp = (1*(sparse_matrix_A==110) * ((sparse_matrix_B==110).transpose())).todense()
+    d = d + n_comp
+
+    temp_total = np.zeros((n_seqs_A, n_seqs_B))
+    temp_total[:] = (1*(sparse_matrix_A>0)).sum(1)
+    temp_total += (1*(sparse_matrix_B>0)).sum(1).transpose()    
+
+    total_differences_shared = (1*(sparse_matrix_A>0)) * (sparse_matrix_B.transpose()>0)
+
+    n_total = np.zeros((n_seqs_A, n_seqs_B))
+    n_sum = (1*(sparse_matrix_A==110)).sum(1)
+    n_total[:] = n_sum
+    n_total += (1*(sparse_matrix_B==110)).sum(1).transpose()  
+
+    diff_n = n_total - 2*n_comp
+    d = temp_total - total_differences_shared.todense() - d - diff_n
+
+    return d
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="generate priorities files based on genetic proximity to focal sample",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument("--alignment", type=str, required=True, help="FASTA file of alignment")
+    parser.add_argument("--metadata", type = str, required=True, help="metadata")
+    parser.add_argument("--focal-alignment", type = str, required=True, help="focal smaple of sequences")
+    parser.add_argument("--output", type=str, required=True, help="FASTA file of output alignment")
+    args = parser.parse_args()
+
+    # load entire alignment and the alignment of focal sequences (upper case -- probably not necessary)
+    context_seqs_dict = calculate_snp_matrix(args.alignment)
+    focal_seqs_dict = calculate_snp_matrix(args.focal_alignment, consensus = context_seqs_dict['consensus'])
+    print("Done reading the alignments.")
+
+    # read in focal alignment to figure out number N & gap
+    focal_seqs = [x.upper() for x in AlignIO.read(args.focal_alignment, format='fasta')]
+    focal_seqs_array = np.array([list(str(x.seq)) for x in focal_seqs])
+    
+    #focal_seqs2 = {x.id:x.upper() for x in AlignIO.read(args.focal_alignment, format='fasta')}
+    #focal_seqs_array2 = np.array([list(str(focal_seqs[x].seq)) for x in focal_seqs])
+    mask_count = np.sum(np.logical_or(focal_seqs_array=='N',focal_seqs_array=='-'), axis=1)
+
+    # remove focal sequences from all sequence to provide context data set
+    keep = [(i, name) for i, name in enumerate(context_seqs_dict['names']) if name not in set(focal_seqs_dict['names'])]
+    context_seqs_dict['snps'] = context_seqs_dict['snps'].tocsr()[[i for i, name in keep],:].tocsc()
+    context_seqs_dict['names'] = [name for i, name in keep]
+
+    # for each context sequence, calculate minimal distance to focal set
+    d = calculate_distance_matrix(context_seqs_dict['snps'], focal_seqs_dict['snps'], consensus = context_seqs_dict['consensus'])
```

Comment:
`calculate_distance_matrix` does not use the `consensus` arg. Not being pedantic, rather want to check that the consensus wasn't intended to be used.

Reply:
Ah, yes this is no longer needed. It is used in the original `pairsnp` package to improve the performance of the algorithm on long bacterial alignments with many gaps. I removed it here to clean up the code but forgot to remove the argument.